### ISSUE TITLE
Request permission result listener of Plugin Registry not called

### ISSUE THREAD

**Issue by rickyazhari** [flutter/flutter#21486]
i try request some permission in my flutter app. The permission request invoking smoothly but there is no result of permission action. i try to add log inside of method onRequestPermissionsResult. It seems not call at all

heres my android porting code : 

```java
@TargetApi(Build.VERSION_CODES.M)
      private void requestMediaPermission(){
        Log.e("permission","request gallery permission.");
        vAct.requestPermissions(new String[]{Manifest.permission.READ_EXTERNAL_STORAGE},102);
    
      }
    
        @Override
          public boolean onRequestPermissionsResult(int i, String[] strings, int[] ints) {
            Log.e("permission"," value i : "+i);
            Log.e("permission"," permission : "+strings[0]);
            Log.e("permission"," permission : "+ints[0]);
            boolean granted = ints.length > 0 && ints[0] == PackageManager.PERMISSION_GRANTED;
            if(granted){
              vChannel.invokeMethod("grantingPermission",null);
            }else{
              if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
                if(vAct.shouldShowRequestPermissionRationale(strings[0])){
                  vChannel.invokeMethod("determinedPermission", null);
                }else{
                  vChannel.invokeMethod("deniedPermission", null);
                }
              }
            }
            return granted;
          }
```

my flutter docor

```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, v0.7.3, on Mac OS X 10.13.4 17E202, locale en-ID)
[✓] Android toolchain - develop for Android devices (Android SDK 27.0.3)
[!] iOS toolchain - develop for iOS devices (Xcode 9.4.1)
    ! CocoaPods out of date (1.5.0 is recommended).
        CocoaPods is used to retrieve the iOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without resolving iOS dependencies with CocoaPods, plugins will not work on iOS.
        For more info, see https://flutter.io/platform-plugins
      To upgrade:
        brew upgrade cocoapods
        pod setup
[✓] Android Studio (version 3.0)
[!] VS Code (version 1.26.1)
[✓] Connected devices (2 available)

! Doctor found issues in 2 categories.
```

this my log after three times try to request permission and deny it

   ```
 E/permission(16414): request gallery permission.
    E/permission(16414): request gallery permission.
    E/permission(16414): request gallery permission.
```

It is running before update beta 0.6.0 .   Please let me know what is actually problem.

**Comment by zoechi** [comment]
That's Java code. That doesn't seem to be Flutter related.
Did you check how it's done in https://pub.dartlang.org/packages/simple_permissions ?

**Comment by zoechi** [comment]
> It is running before update beta 0.6.0 . 

Uh, I missed that line. So this code already worked?

**Comment by rickyazhari** [comment]
yah the listener called in v 0.5.x

**Comment by jgerryl** [comment]
Hello, I'm experiencing the same problem. the boolean onRequestPermissionsResult() doesn't get called. 
Is there any update on this issue? I'm using flutter v.0.9.4.
Thanks!

**Comment by jgerryl** [comment]
Hello, I just upgraded to flutter v.1.0.0 but the problem still exists. Any help would be really appreciated. Thanks!

**Comment by jgerryl** [comment]
Hello @zoechi,
I just tried using [simple_permissions](https://pub.dartlang.org/packages/simple_permissions) plugin to check whether the onRequestPermissionsResults() got called after requesting permission in Android emulator, and it does! (I tried with the demo app in the plugin github).
So I checked the plugin source code [here](https://github.com/Ethras/flutter_simple_permissions/blob/master/android/src/main/java/com/ethras/simplepermissions/SimplePermissionsPlugin.java) to see what's different with the image_picker plugin.

I found that in image_picker plugin, the addRequestPermissionsResultListener is set to ImagePickerDelegate which is in a separate file than ImagePickerPlugin.java where it is declared.
I tried checking the activity used in ImagePickerDelegate when requesting permission, and it points to my app's MainActivity (same as in simple_permissions plugin), but the onRequestPermissionsResults() just don't get called in ImagePickerDelegate.

Do you have any opinion about this? Should I try putting the process in ImagePickerDelegate into ImagePickerPlugin to see if it works? Please help. Thanks!

**Comment by jgerryl** [comment]
Hello @zoechi,
Today, I found that the permission request in image_picker plugin did call onRequestPermissionsResults(). But, the one called is the onRequestPermissionsResults() in my app's MainActivity (this is my additional code) and not the one in ImagePickerDelegate.
Is this they way it should be? Does that mean I can't add additional permission request in my MainActivity for other permission related process or is there any other way this could be accomplished?
Please help, thanks!

**Comment by jgerryl** [comment]
oh and @zoechi, just to be clear. 
I use native method for onRequestPermissionsResults on my app's `MainActivity`:
```
@Override
  public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {}
```

while the image_picker plugin uses the one from the PluginRegistry.RequestPermissionsResultListener
```
@Override
   public boolean onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {}
```

Thanks!

**Comment by jgerryl** [comment]
Dear @zoechi,
in the end I just made a plugin to handle my additional permission related process. thanks!

**Comment by rickyazhari** [comment]
@jgerryl  YOLO

**Comment by collinjackson** [comment]
I don't think there's action here for the plugins team, removing the plugins label to get it off our triage list.

**Comment by ftcvlad** [comment]
the same issue is happening in flutter_qrcode_reader plugin.  [Source](https://github.com/Kumulos/flutter_qrcode_reader/blob/master/android/src/main/java/com/matheusvillela/flutter/plugins/qrcodereader/QRCodeReaderPlugin.java)

`boolean onRequestPermissionsResult ` is not getting called

**Comment by ftcvlad** [comment]
@jgerryl 

> is there any other way this could be accomplished?

yes, in your `MainActivity` you can do

@Override
    public void onRequestPermissionsResult(int requestCode, String[] permissions, int[] grantResults) {
        if (requestCode == REQUEST_PERMISSIONS_REQUEST_CODE) {
            ....
        }
        else{
            super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        }
    }

this way both `onRequestPermissionsResult` are executed

**Comment by dolphinxx** [comment]
I also encountered this issue today. After an hour of investigation, I finally found out the cause of the problem.
I have two plugins that both implement the method for requesting permissions. One of the plugin's onRequestPermissionsResult method always returns true ignoring the requestCode, causing other plugins to no longer receive the result of this request.
So we should choose a unique requestCode, and return true only receiving the expected requestCode in plugins.
